codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -964,6 +1020,14 @@ protected ValueConverter<V> valueConverter() {
         return valueConverter;
     }
 
+    protected NameValidator<K> nameValidator() {
```

Comment:
I see that this is required for HTTP/2 headers impl. Instead of exposing these 2 methods as `protected` API consider caching the `nameValidator` in `DefaultHttp2Headers` instead of requiring this method. That way we can minimize public API surface.

Reply:
I think that these being hidden is inconsistent with the existing `valueConverter()`, and such inconsistencies lead to worse code. For instance, to not have this method available would mean duplicating the reference to the name validator, as you say. I think instead the `DefaultHeaders` class is designed to be extended, and these methods is part of making the design whole.